cognite/model_hosting/data_fetcher/_client/api_client.py

```diff
@@ -17,17 +17,26 @@
 DEFAULT_BASE_URL = "https://api.cognitedata.com"
 NUM_OF_RETRIES = int(os.getenv("COGNITE_DATA_FETCHER_RETRIES", 3))
 TIMEOUT = int(os.getenv("COGNITE_DATA_FETCHER_TIMEOUT", 10))
+BACKOFF_MAX = 30
 
 HTTP_STATUS_CODES_TO_RETRY = [429, 500, 502, 503]
-HTTP_METHODS_TO_RETRY = ["GET", "POST", "DELETE"]  # We can retry POST because we only fetch data
 
 log = logging.getLogger("data-fetcher")
 
 
+class RetryWithMaxBackoff(Retry):
+    def get_backoff_time(self):
+        return min(BACKOFF_MAX, super().get_backoff_time())
+
+
 def _init_requests_session():
     session = _requests.Session()
-    retry = Retry(
-        total=NUM_OF_RETRIES, backoff_factor=0.3, status_forcelist=HTTP_STATUS_CODES_TO_RETRY, raise_on_status=False
+    retry = RetryWithMaxBackoff(
```

Comment:
can't you just set `retry.BACKOFF_MAX = BACKOFF_MAX` instead of subclassing Retry?

Reply:
It's a class property, so that would set it globally and potentially be very confusing and disturb other uses of (the very) common urllib3 library in the same process.